File: Algorithms/src/2033. Minimum Operations to Make a Uni-Value Grid/Code/solution.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        nums = []

        # Flatten grid
        for row in grid:
            nums.extend(row)

        # Check feasibility
        rem = nums[0] % x
        for num in nums:
            if num % x != rem:
                return -1

        # Sort
        nums.sort()

        # Median
        median = nums[len(nums) // 2]

        # Count operations
        ops = 0
        for num in nums:
            ops += abs(num - median) // x

        return ops

File: Algorithms/src/2033. Minimum Operations to Make a Uni-Value Grid/Code/solution.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        List<Integer> nums = new ArrayList<>();

        // Flatten grid
        for (int[] row : grid) {
            for (int val : row) {
                nums.add(val);
            }
        }

        // Check feasibility
        int rem = nums.get(0) % x;
        for (int num : nums) {
            if (num % x != rem)
                return -1;
        }

        // Sort
        Collections.sort(nums);

        // Median
        int median = nums.get(nums.size() / 2);

        // Count operations
        int ops = 0;
        for (int num : nums) {
            ops += Math.abs(num - median) / x;
        }

        return ops;
    }
}